adoptingerlang.org

```diff
@@ -1334,6 +1334,370 @@ Repeat this exercise from time to time, and eventually do some [[https://princip
+
+As such, Rebar3 has the following properties:
+
+- Rebar3 is a declarative build tool. You provide a configuration, and it executes code to satisfy that configuration. If you want to run custom scripts and extend the build, it specifically provides [[https://www.rebar3.org/docs/configuration#section-hooks][hooks]] or [[https://www.rebar3.org/docs/using-available-plugins][plugins]] interfaces to do so. There are also ways to gain [[https://www.rebar3.org/docs/dynamic-configuration][Dynamic Configuration]] to more flexibly populate config files.
+- As part of that declarative approach, commands in Rebar3 are defined with dependency sequences. For example, the =rebar3 compile= task relies on =rebar3 get-deps= (or rather, a private form of it that locks dependencies), and will run it for you. The command =rebar3 tar= will implicitly call a sequence that includes =get-deps -> compile -> release -> tar= for you. Rebar3 therefore knows that to compile your own project, its dependencies are needed.
+- Rebar3 defines its own working area within the =_build= directory; it expects to have control over what goes in there, and for you not to track that directory in source control nor rely on its internal structure. When it builds a thing you want to use, it usually outputs the path in the terminal. For example =rebar3 escriptize= will print the path to the generated escript in =_build/default/bin/= after each run.
+- All projects are built locally within their own directory based on the Erlang runtime currently loaded in your environment
+- All dependencies are fetched to the project's =_build= directory
+- All dependencies can define their own dependencies, and rebar3 will recognize this and put them within your own project's =_build= directory as well
+- Since version numbers are unreliable (even with semantic versioning), we consider versions are information for humans, not build tools. In case of libraries that end up declared multiple times, the one closest to the project root is chosen to win ([[https://www.rebar3.org/docs/dependencies][warnings]] are emitted during the first build to let the user know about this), with the assumption that libraries declared closer to the root project are used more thoroughly (and in the case of semver issues, the root project has to pick a winner anyway, which is possible here)
```

Comment:
```suggestion
- Since version numbers are unreliable (even with semantic versioning), we consider versions as information for humans, not build tools. In case of libraries that end up declared multiple times, the one closest to the project root is chosen, while [[https://www.rebar3.org/docs/dependencies][warnings]] are emitted during the first build (it knows it is the "first" because there is no lock for the dependency) to let the user know versions not being used, with the assumption that libraries declared closer to the root project are used more thoroughly
```

I don't understand the last part about semver, maybe just needs rewording.